src/fluree/json_ld/impl/context.cljc

```diff
@@ -2,12 +2,76 @@
   (:require [fluree.json-ld.impl.iri :as iri]
             [fluree.json-ld.impl.util :refer [try-catchall]]
             [fluree.json-ld.impl.external :as external]
+            [fluree.json-ld.impl.compact :as compact]
             [clojure.string :as str]))
 
 #?(:clj (set! *warn-on-reflection* true))
 
 (declare parse)
 
+(defn- stringify-ctx-kw
+  "If an item is a keyword, turns it back into a string with '@'
+  e.g. :id -> @id, :type -> @type"
+  [kw]
+  (if (keyword? kw)
+    (str "@" (name kw))
+    kw))
+
+
+(defn un-parse
+  "Takes the parsed context turns it back into a single map context.
+
+  This is designed to be as close to the likely provided context as possible
+  and sacrifices some speed to achieve that."
+  [{:keys [type-key] :as parsed-context}]
+  (reduce-kv
+    (fn [acc k v]
+      (cond
+        (= :type-key k)
+        acc
+
+        ;; just a single map entry with an :id
+        (and (= 1 (count v))
+             (get v :id))
+        (assoc acc k (:id v))
+
+        (and (= type-key k)
+             (= 2 (count v)))
+        (assoc acc k "@type")
+
+        :else
+        (let [v* (reduce-kv
+                   (fn [acc' k' v']
+                     (cond
+
+                       (= k' :id)
+                       (cond
```

Comment:
I think we can add an `alias?` fn and use it here to clarify the intent a bit more:
```clojure
(defn alias?
  [x]
  (or (and (string? x)
           (str/includes? x ":"))
      (and (keyword? x)
           (-> x namespace boolean))))
```